anaconda.py

```diff
@@ -1095,26 +591,30 @@ def _earlyExceptionHandler(ty, value, traceback):
     screen_access.sam.open_config_file()
 
     # now start the interface
-    setupDisplay(anaconda, opts, addon_paths)
+    display.setup_display(anaconda, opts, addon_paths=addon_paths)
     if anaconda.gui_startup_failed:
         # we need to reinitialize the locale if GUI startup failed,
         # as we might now be in text mode, which might not be able to display
         # the characters from our current locale
         log.warning("reinitializing locale due to failed attempt to start the GUI")
-        localization.setup_locale(os.environ["LANG"], ksdata.lang, anaconda.displayMode != "g")
+        localization.setup_locale(os.environ["LANG"], ksdata.lang, text_mode=anaconda.tui_mode)
 
     # we now know in which mode we are going to run so store the information
-    from pykickstart.constants import DISPLAY_MODE_GRAPHICAL, DISPLAY_MODE_CMDLINE, DISPLAY_MODE_TEXT
-    mode_char_to_const = {'g': DISPLAY_MODE_GRAPHICAL, 't': DISPLAY_MODE_TEXT, 'c': DISPLAY_MODE_CMDLINE}
-    ksdata.displaymode.displayMode = mode_char_to_const[anaconda.displayMode]
+    from pykickstart import constants as pykickstart_constants
+    mode_char_to_const = {
```

Comment:
There should be used some other name. The variable ``mode_char_to_const`` doesn't give much sense now.
What about ``display_mode_convertor``.